### PR TITLE
Fix VCI comm hints problems

### PR DESCRIPTION
**Pull request by sagarth**
## Pull Request Description
This PR
* fixes a bug related to VCI comm hints assignment
* corrects where the CH4 hints are registered

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

File: src/mpid/ch4/src/ch4_init.c [hunk 1]
```diff
@@ -300,6 +300,20 @@ static int set_runtime_configurations(void)
 #error "Thread Granularity:  Invalid"
 #endif
 
+static void register_comm_hints(void);
```
Comment (hzhou) [inline src/mpid/ch4/src/ch4_init.c:303]: There is no need to declare `static` functions.

File: src/mpid/ch4/src/ch4_init.c [hunk 2]
```diff
@@ -455,6 +469,9 @@ int MPID_Init(int requested, int *provided)
     MPIR_Process.attrs.appnum = MPIR_Process.appnum;
     MPIR_Process.attrs.io = MPI_ANY_SOURCE;
 
+    /* Initialize comm hints */
```
Comment (hzhou) [inline src/mpid/ch4/src/ch4_init.c:472]: This comment doesn't provide more than the function name already says. Suggest remove.

File: src/mpid/ch4/src/ch4_init.c [hunk 3]
```diff
@@ -300,6 +300,20 @@ static int set_runtime_configurations(void)
 #error "Thread Granularity:  Invalid"
 #endif
 
+static void register_comm_hints(void);
+
+/* Register CH4-specific hints */
+static void register_comm_hints(void)
+{
+    /* Non-standard hints for VCI selection. Default value MPIDI_VCI_INVALID (-1) */
```
Comment (hzhou) [inline src/mpid/ch4/src/ch4_init.c:308]: Why not replace the `-1` below with `MPIDI_VCI_INVALID` (so we don't need comment for it)?

**Review by hzhou — COMMENTED** [review]
Looks good. Some nitpick.

**Comment by sagarth** [comment]
@hzhou Addressed your recent comments.

**Comment by sagarth** [comment]
test:mpich/custom
netmod: ch4:ofi
config: vci

**Review by hzhou — APPROVED** [review]
LGTM.

@sagarth Could you confirm it fixes the issue on your end?

**Comment by sagarth** [comment]
@hzhou Just verified with local testing that this PR fixes the issue.

**Pull request merged**